SQF/dayz_code/compile/doorManagement/fn_check_access.sqf

```diff
@@ -0,0 +1,146 @@
+	,"_isPlotOwner"					// return value
+	,"_isPlotFriend"				// return value
+	,"_isPlotAdmin"					// return value
+	,"_isTargetFriend"				// return value
+	,"_isTargetAdmin"				// return value
+	,"_targetType"					// return value
+	,"_targetOwnerUID"				// UID of the owner of _target
+	,"_playerUID"					// UID of the _player
+	,"_plotcheck"					// takes return value of FNC_find_plots
+	,"_isNearPlot"					// player is in plot's reach
+	,"_nearestPlot"					// plot object
+	,"_plotOwnerUID"				// plot owner's UID
+	,"_plotFriends"					// list of plot friends [["UID", "Name"], ..]
+	,"_targetFriends"				// list of target friends [["UID", "Name"], ..]
+];
+
+
+// assign parameters
+_player = _this select 0;
+_target = _this select 1;
+
+// Initialize return values
+_isOwner = false;
+_isFriendly = false;
+_isPlotOwner = false;
+_isPlotFriend = false;
+_isPlotAdmin = false;
+_isTargetFriend = false;
+_isTargetAdmin = false;
+_targetType = "";
+
+
+// determine target type
+_targetType = if(typeOf _target in DZE_DoorsLocked) then { "DOOR"; } else { "GENERIC"; };
+
+
+// determine owner and player id
+// and check if player is owner of target object
+if(DZE_permanentPlot) then {
+	_targetOwnerUID = _target getVariable ["ownerPUID","0"];
+	_playerUID = [_player] call FNC_GetPlayerUID;
+} else {
+	_targetOwnerUID = _target getVariable ["characterID","0"];
+	_playerUID = dayz_characterID;
+};
+_isOwner = (_playerUID == _targetOwnerUID);
+
+
+// determine _players friends (tagged)
+// and check if owner of _target is tagged friendly
+_friendlies	= _player getVariable ["friendlyTo",[]];
+_isFriendly = (_targetOwnerUID in _friendlies);
+
+
+// find nearest plot
+_plotcheck = [_player, false] call FNC_find_plots;
+_isNearPlot = ((_plotcheck select 1) > 0);
+_nearestPlot = _plotcheck select 2;
+if(_isNearPlot) then {
+	// determine plot owner
+	// and check if player is owner of plot
+	if (DZE_permanentPlot) then {
+		_plotOwnerUID = _nearestPlot getVariable ["ownerPUID","0"];
+	} else {
+		_plotOwnerUID = _target getVariable ["characterID","0"];
+	};
+	_isPlotOwner = (_playerUID == _plotOwnerUID);
+	
+	
+	// determine plot friends
+	// and check if player is one of them
+	_isPlotFriend = (_playerUID == _plotOwnerUID);  // Plot owner is always a plot friend
+	_plotFriends = _nearestPlot getVariable ["plotfriends", []];
+	{
+		if( (_x select 0) == _playerUID ) then { _isPlotFriend = true; };
+	} count _plotFriends;
+	
+	
+	// determine plot management admins
+	// and check if player is one of them
+	{
+		if( _x == _playerUID ) then { _isPlotAdmin = true; };
+	} count DZE_PlotManagementAdmins;
+};
+
+
+// Process target type DOOR
+if(_targetType == "DOOR") then {
+	// determine door friends
+	// and check if player is one of them
+	_isTargetFriend = (_playerUID == _targetOwnerUID);  // Door owner is always a door friend
+	_targetFriends = _target getVariable ["doorfriends",[]];
+	{
+		if( (_x select 0) == _playerUID ) then { _isTargetFriend = true; };
```

Comment:
Since you changed _playerUID into characterID when DZE_permanentPlot = false; above you need to have a separate variable like `_realPlayerUID = [_player] call FNC_GetPlayerUID;` at the top of this file. Then change this to 
```sqf
_if( (_x select 0) == _realPlayerUID ) then {
```
Otherwise you are breaking door management when DZE_permanentPlot = false;, because door management never saves characterIDs.